src/cmap/connection.ts

```diff
@@ -1171,26 +1009,130 @@ export class ModernConnection extends TypedEventEmitter<ConnectionEvents> {
       };
     }
 
-    const commandOptions: Document = Object.assign(
-      {
-        numberToSkip: 0,
-        numberToReturn: -1,
-        checkKeys: false,
-        // This value is not overridable
-        secondaryOk: readPreference.secondaryOk()
-      },
-      options
-    );
+    const commandOptions = {
+      numberToSkip: 0,
+      numberToReturn: -1,
+      checkKeys: false,
+      // This value is not overridable
+      secondaryOk: readPreference.secondaryOk(),
+      ...options,
+      readPreference // ensure we pass in ReadPreference instance
+    };
 
     const message = this.supportsOpMsg
-      ? new OpMsgRequest(ns.db, cmd, commandOptions)
-      : new OpQueryRequest(ns.db, cmd, commandOptions);
+      ? new OpMsgRequest(db, cmd, commandOptions)
+      : new OpQueryRequest(db, cmd, commandOptions);
+
+    return message;
+  }
+
+  private async sendCommand(
+    message: WriteProtocolMessageType,
+    options: CommandOptions
+  ): Promise<Document> {
+    const { signal } = this.controller;
 
+    signal.throwIfAborted();
```

Comment:
i commented on another `throwIfAborted()`, but I'm surprised we need them in every helper?  My thought was that it should be sufficient to only check when we actually read/write to/from the socket

Reply:
In this specific case I could remove this and because of how we have the io helpers set up there will be no difference.

This might be a practices thing we want to discuss in tech chat. I believe the thinking is that you want to check for aborted ahead of any `await` for at least two reasons, 1. you may not be able to trust the API taking the signal to handle an already aborted signal, so you may block forever as a result. 2. You yeild to the queue of tasks despite having indication of an error already.

Kinda thinking/talking off the cuff here so I'll do some research and gather thoughts for chat.

Reply:
curious about best practices.  I assumed that the best practice would be to just ferry the abort controller along to whatever async function we're calling and letting them handle it, unless there's a strong reason to manually check it ourselves

Reply:
https://openjsf.org/blog/2022/04/29/using-abortsignal-in-node-js/

This was illuminating on the subject of signals

> Within this function, you should immediately check to see if the signal has already been triggered and, if it has, immediately abort the operation.

And this makes sense: if you support signals, before doing any work like setting up listeners and kicking off tasks you want to bail out before performing all that labor.